clean-data/clean-data.R

```diff
@@ -0,0 +1,700 @@
+        mutate_each(funs(grepl(findStr, ., ignore.case = TRUE))) %>%
+        unlist(use.names = FALSE)
+    subDirty <- dirtyDat %>% filter(dirtyIdx) %>%
+        mutate_(.dots = setNames(list(varval), colName))
+
+    cleanDat <- dirtyDat %>% filter(!dirtyIdx) %>% bind_rows(subDirty)
+
+    cleanDat
+}
+
+# Clean Expected Earnings
+clean_expected_earnings <- function(cleanPart1) {
+    # Remove dollar signs from expected earnings
+    dollarIdx <- cleanPart1 %>% select(ExpectedEarning) %>%
+        mutate_each(funs(grepl("\\$", ., ignore.case = TRUE))) %>%
+        unlist(use.names = FALSE)
+    dollarData <- cleanPart1 %>% filter(dollarIdx) %>%
+        mutate(ExpectedEarning = sub("\\$", "", ExpectedEarning))
+    cleanPart1 <- cleanPart1 %>% filter(!dollarIdx) %>% bind_rows(dollarData)
+
+    # Remove commas from expected earnings
+    commaIdx <- cleanPart1 %>% select(ExpectedEarning) %>%
+        mutate_each(funs(grepl(",", ., ignore.case = TRUE))) %>%
+        unlist(use.names = FALSE)
+    commaData <- cleanPart1 %>% filter(commaIdx) %>%
+        mutate(ExpectedEarning = sub(",", "", ExpectedEarning))
+    cleanPart1 <- cleanPart1 %>% filter(!commaIdx) %>% bind_rows(commaData)
+
+    # Remove "k" from expected earnings
+    kIdx <- cleanPart1 %>% select(ExpectedEarning) %>%
+        mutate_each(funs(grepl("k", ., ignore.case = TRUE))) %>%
+        unlist(use.names = FALSE)
+    kData <- cleanPart1 %>% filter(kIdx) %>%
+        mutate(ExpectedEarning = sub("k", "000", ExpectedEarning))
+    cleanPart1 <- cleanPart1 %>% filter(!kIdx) %>% bind_rows(kData)
+
+    # Change range of expected earnings into average
+    rangeIdx <- cleanPart1 %>% select(ExpectedEarning) %>%
+        mutate_each(funs(grepl("-", ., ignore.case = TRUE))) %>%
+        unlist(use.names = FALSE)
+    rangeData <- cleanPart1 %>% filter(rangeIdx) %>%
+        mutate(ExpectedEarning = average_range_earning(ExpectedEarning))
+    cleanPart1 <- cleanPart1 %>% filter(!rangeIdx) %>% bind_rows(rangeData)
+
+    # Remove period from salaries like 50.000 which should be just 50000
+    thousandsIdx <- cleanPart1 %>% select(ExpectedEarning) %>%
+        mutate_each(funs(grepl("^\\d{2}\\.", ., ignore.case = TRUE))) %>%
+        unlist(use.names = FALSE)
+    thousandsData <- cleanPart1 %>% filter(thousandsIdx) %>%
+        mutate(ExpectedEarning = sub("\\.", "", ExpectedEarning))
+    cleanPart1 <- cleanPart1 %>% filter(!thousandsIdx) %>%
+        bind_rows(thousandsData)
+
+    # Remove any non-numeric characters
+    numericIdx <- cleanPart1 %>% select(ExpectedEarning) %>%
+        mutate_each(funs(grepl("[A-Za-z]", ., ignore.case = TRUE))) %>%
+        unlist(use.names = FALSE)
+    numericData <- cleanPart1 %>% filter(numericIdx) %>%
+        mutate(ExpectedEarning = gsub("[A-Za-z']*", "", ExpectedEarning))
+    cleanPart1 <- cleanPart1 %>% filter(!numericIdx) %>%
+        bind_rows(numericData)
+
+    # Make all expected earnings less than 100 multiplied by 1000
```

Comment:
@erictleung no problem, I like data :)

Firstly, we all should **agree** on one thing.
Should we just cutoff (into blank) all weird numbers or should we guess what was the real intention and then try to normalize it? And it's not only about expected salary, but about all questions with open answers with numbers. @erictleung @evaristoc @QuincyLarson